### PR TITLE
Created a seperate function in order to unmarshal

### PR DESCRIPTION
**Pull request by kmada97**
Signed-Off-By: Kalyan Mada <kalyan.red.mada@hpe.com>

### CODE REVIEW

File: sources/pmwebapi.go [hunk 1]
```diff
@@ -103,25 +103,26 @@ func getRequest(url string) []byte {
...
 	contextNumber := strconv.FormatFloat(tokenMap["context"], 'f', -1, 64)
 	getMetricsBody := getRequest("http://localhost:44323/pmapi/" + contextNumber + "/_metric")
 
 	metricHeaderVar := &MetricHeader{}
-	err = json.Unmarshal([]byte(getMetricsBody), metricHeaderVar)
-	if err != nil {
-		log.Fatal("NewRequest: ", err)
-	}
+	unmarshal([]byte(getMetricsBody), metricHeaderVar)
+
```
Comment (roclark) [inline sources/pmwebapi.go]: Looks like there's an extraneous line here - I'd remove it before merging.

**Review by roclark — COMMENTED** [review]
Good job moving this into another function - this makes a big difference and you will thank yourself when it comes to unit testing. Only item I see is the extra line. Once that's taken care of I will approve.

File: sources/pmwebapi.go (hunk 1 above)
Comment (joehandzik) [reply sources/pmwebapi.go]: agreed

**Comment by joehandzik** [comment]
While I will merge this, it's not exactly the sort of refactor I would target aggressively. You've saved some lines of code, but perhaps at the expense of making it easy and fast for someone to see that you're checking errors for that operation. Does that make sense? Any thoughts from that angle, @roclark?

**Comment by roclark** [comment]
I certainly agree with you @joehandzik, but I think in this case, it is good to move this to a separate function in preparation for the unit tests. I believe that was @kmada97's goal with this PR. Maybe this would be better suited with the future unit test PR, but it still gets the job done in my mind.

**Pull request merged**